### PR TITLE
Create basic architecture of variational autoencoder

### PR DESCRIPTION
**Pull request by geek-yang**
In this PR, we create the basic structure of variational autoencoder. The encoder and decoder parts are included in one class.

### CODE REVIEW

**Comment by geek-yang** [comment]
This PR will close #9  #10 and #11.

File: gait_mapper/vae.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import tensorflow as tf
+from gait_mapper import utils
+
+
+class VAE(tf.keras.Model):
+    """Variational autoencoder of the gait mapper."""
+
+    def __init__(self, window=200, dof=6, latent_features=6, alpha=0.1):
```
Comment (CunliangGeng) [inline gait_mapper/vae.py:8]: Docstrings are needed for these parameters

File: gait_mapper/utils.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import tensorflow as tf
+
+
+def sample_latent_features(distribution):
```
Comment (CunliangGeng) [inline gait_mapper/utils.py:4]: The sampling function is used only by VAE class, it's also fine to put it to the class as an private method

**Review by CunliangGeng — CHANGES_REQUESTED** [review]
Thanks Yang, please check the two minor comments above.

**Comment by CunliangGeng** [comment]
A minor comment about the notebook tutorial: usually we don't use `mixedCase` for variable name in python, better to use `lowercase` or `lower_case`, see detail in pep8 https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

File: gait_mapper/vae.py (hunk 1 above)
Comment (geek-yang) [reply gait_mapper/vae.py]: Good point! Corrected!

**Comment by geek-yang** [comment]
> A minor comment about the notebook tutorial: usually we don't use `mixedCase` for variable name in python, better to use `lowercase` or `lower_case`, see detail in pep8 https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Thanks for pointing this out! I rename all these variables using lowercase now.

File: gait_mapper/utils.py (hunk 2 above)
Comment (geek-yang) [reply gait_mapper/utils.py:4]: Corrected!

**Pull request merged**